src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2216,7 +2216,7 @@ public static void Select(IList checkRead, IList checkWrite, IList checkError, i
             // Validate input parameters.
             if ((checkRead == null || checkRead.Count == 0) && (checkWrite == null || checkWrite.Count == 0) && (checkError == null || checkError.Count == 0))
             {
-                throw new ArgumentNullException(SR.net_sockets_empty_select);
+                throw new ArgumentNullException(null, SR.net_sockets_empty_select);
```

Comment:
+1 on @hughbe's answer: the API needs at least one of the params to be not-null or empty: https://msdn.microsoft.com/en-us/library/system.net.sockets.socket.select(v=vs.110).aspx